src/Bundle/Maker/Factory/AbstractDoctrineDefaultPropertiesGuesser.php

```diff
@@ -1,46 +1,54 @@
 <?php
 
-declare(strict_types=1);
-
 namespace Zenstruck\Foundry\Bundle\Maker\Factory;
 
 use Doctrine\Persistence\ManagerRegistry;
 use Doctrine\Persistence\Mapping\ClassMetadata;
-use Doctrine\Persistence\ObjectManager;
+use Symfony\Component\Console\Style\SymfonyStyle;
 
 /** @internal  */
 abstract class AbstractDoctrineDefaultPropertiesGuesser implements DefaultPropertiesGuesser
 {
-    public function __construct(protected ManagerRegistry $managerRegistry, private FactoryFinder $factoryFinder)
+    public function __construct(protected ManagerRegistry $managerRegistry, private FactoryFinder $factoryFinder, private FactoryGenerator $factoryGenerator)
     {
     }
 
     /** @param class-string $fieldClass */
-    protected function addDefaultValueUsingFactory(MakeFactoryData $makeFactoryData, string $fieldName, string $fieldClass, bool $isMultiple = false): void
+    protected function addDefaultValueUsingFactory(SymfonyStyle $io, MakeFactoryData $makeFactoryData, MakeFactoryQuery $makeFactoryQuery, string $fieldName, string $fieldClass, bool $isMultiple = false): void
     {
         if (!$factoryClass = $this->factoryFinder->getFactoryForClass($fieldClass)) {
-            $makeFactoryData->addDefaultProperty(\lcfirst($fieldName), "null, // TODO add {$fieldClass} type manually");
+            if ($makeFactoryQuery->isAllFields() || $io->confirm("A factory for class \"{$fieldClass}\" is missing for field {$makeFactoryData->getObjectShortName()}::\${$fieldName}. Do you want to create it?")) {
+                $factoryClass = $this->factoryGenerator->generateFactory($io, $makeFactoryQuery->withClass($fieldClass));
+            } else {
+                $makeFactoryData->addDefaultProperty(\lcfirst($fieldName), "null, // TODO add {$fieldClass} type manually");
 
-            return;
+                return;
+            }
         }
 
         $factoryMethod = $isMultiple ? 'new()->many(5)' : 'new()';
```

Comment:
Weren't we going to remove this?

Reply:
we removed it for "multiple" relationship (ie: `{One|Many}ToMany`)

but we could still have [multiple `EmbeddedDocument` for ODM](https://github.com/zenstruck/foundry/blob/1.x/tests/Fixtures/Document/Post.php#L49-L54). There is no risk to have circular dependencies here.

If you prefer, I can remove this one, since it's a collection, it is not mandatory

Reply:
Ah ok, maybe we should remove just for consistency?

Reply:
But don't worry about it for this PR.

Reply:
already resolved :sweat_smile: